src/main/java/org/seqdoop/hadoop_bam/VCFRecordReader.java

```diff
@@ -126,7 +163,7 @@
 	@Override public VariantContextWritable getCurrentValue() { return vc; }
 
 	@Override public boolean nextKeyValue() throws IOException {
-		if (!it.hasNext() || it.getPosition() >= length)
+		if (!it.hasNext() || getPosition() > length)
```

Comment:
Note the boundary change here, which was needed to get the test for test.vcf.gz to pass. I think it's needed because the underlying stream buffers, so it reads to the end of the file before finishing the iteration. Note that this condition is now the same as the one above (line 115/152), so it should be OK.